flake.nix

```
@@ -22,135 +28,142 @@
       url = "github:edolstra/flake-compat";
       flake = false;
     };
-    flake-utils.url = "github:numtide/flake-utils";
   };
 
-  outputs = { self, nixpkgs, nixpkgs-test, crane, rust-overlay, ... }:
-    let
-      pkgs = import nixpkgs {
-        system = "x86_64-linux";
-        overlays = [
-          rust-overlay.overlays.default
-        ];
-      };
-
-      testPkgs = import nixpkgs-test { system = "x86_64-linux"; };
+  outputs = inputs@{ self, nixpkgs, nixpkgs-test, crane, rust-overlay, flake-parts, ... }:
+    flake-parts.lib.mkFlake { inherit inputs; } ({ moduleWithSystem, ... }: {
+      imports = [
+        # Derive the output overlay automatically from all packages that we define.
+        inputs.flake-parts.flakeModules.easyOverlay
+      ];
 
-      inherit (pkgs) lib;
+      flake.nixosModules.lanzaboote = moduleWithSystem (import ./nix/modules/lanzaboote.nix);
```

Something like this would be better:

```suggestion
     nixosModules.lanzaboote = moduleWithSystem (
       perSystem@{ config }:
       { pkgs, lib, ... }: {
          imports = [ ./nix/modules/lanzaboote.nix ];
          boot.lanzaboote.package = lib.mkDefault perSystem.config.packages.${pkgs.system}.lanzatool;
        };
      );
```
 I haven't checked if this is actually valid. This follows the same pattern Robert uses in the example in his docs: https://flake.parts/module-arguments.html#modulewithsystem

---

Fixed.